### PR TITLE
CAMEL-15858: Adds feature for remote file handling for OpenAPI specifications

### PR DESCRIPTION
**Pull request by ex324**
This pull request adds support for using remote OpenAPI specifications.

The implementation uses the existing _specificationUri_ attribute of the plugin allowing the usage of http-URLs additionaly to only looking in the local file system.
The implementation additionally adds a new _auth_ attribute to pass an authorization header. It's usage is added to the plugin documentation.

The code is copied from the OpenAPITools (https://github.com/OpenAPITools/openapi-generator) **openapi-generator-maven-plugin**, so the implementation adds a new maven dependency for the _openapi-generator_.

### CODE REVIEW

File: tooling/maven/camel-restdsl-openapi-plugin/pom.xml [hunk 1]
```diff
@@ -55,6 +55,11 @@
             <artifactId>snakeyaml</artifactId>
             <version>${snakeyaml-version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.openapitools</groupId>
+            <artifactId>openapi-generator</artifactId>
+            <version>4.3.1</version>
```
Comment (oscerd) [inline tooling/maven/camel-restdsl-openapi-plugin/pom.xml:61]: This can be a placeholder for property version in parent/pom

**Review by oscerd — CHANGES_REQUESTED** [review]
One finding, and please amend the commit message with a reference to the camel issue number CAMEL-15858, thanks.

File: tooling/maven/camel-restdsl-openapi-plugin/pom.xml (hunk 1 above)
Comment (ex324) [reply tooling/maven/camel-restdsl-openapi-plugin/pom.xml:61]: Hi. Thanks for the input. Would you like the according property in the maven-plugins pom or in another, e.g. tooling-parent pom?

Comment (oscerd) [reply tooling/maven/camel-restdsl-openapi-plugin/pom.xml:61]: Morning. I meant here: https://github.com/apache/camel/blob/master/parent/pom.xml

Comment (oscerd) [reply tooling/maven/camel-restdsl-openapi-plugin/pom.xml:61]: Thanks

**Review by oscerd — APPROVED** [review]

File: tooling/maven/camel-restdsl-openapi-plugin/src/main/java/org/apache/camel/maven/generator/openapi/AbstractGenerateMojo.java [hunk 2]
```diff
@@ -33,6 +40,7 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 import io.apicurio.datamodels.Library;
 import io.apicurio.datamodels.openapi.models.OasDocument;
+import io.swagger.v3.parser.core.models.AuthorizationValue;
```
Comment (davsclaus) [inline tooling/maven/camel-restdsl-openapi-plugin/src/main/java/org/apache/camel/maven/generator/openapi/AbstractGenerateMojo.java:43]: Can we avoid swagger imports, as we should ideally only use the datamodel from apicurio.

**Comment by oscerd** [comment]
This is still pending.

**Comment by oscerd** [comment]
I think you need to rebase.

File: tooling/maven/camel-restdsl-openapi-plugin/src/main/java/org/apache/camel/maven/generator/openapi/AbstractGenerateMojo.java (hunk 2 above)
Comment (ex324) [reply tooling/maven/camel-restdsl-openapi-plugin/src/main/java/org/apache/camel/maven/generator/openapi/AbstractGenerateMojo.java]: I removed the openapi / swagger dependencies...

**Review by davsclaus — APPROVED** [review]

**Review by oscerd — APPROVED** [review]

**Comment by oscerd** [comment]
LGTM, lets wait for @davsclaus

**Pull request merged**